api/tests/unit/handlers/changes.js

```diff
@@ -94,6 +94,19 @@ exports['filters the changes to relevant ones'] = function(test) {
       { id: allowedId, key: 'subjectId', value: { submitter: 'contactId' } }
     ]
   });
+  // returns the list of subjects the user is allowed to see
+  db.medic.view.onCall(2).callsArgWith(3, null, {
+    rows: [
+      { id: subjectId, value: patientId }
+    ]
+  });
+  // returns the list of doc ids the user is allowed to see
+  db.medic.view.onCall(3).callsArgWith(3, null, {
+    rows: [
+      { id: unchangedId, key: 'subjectId', value: { submitter: 'contactId' } },
+      { id: allowedId, key: 'subjectId', value: { submitter: 'contactId' } }
+    ]
+  });
```

Comment:
These tests are a code smell - we need to break the changes handler up into multiple modules so we can unit test each separately rather than having to mock the whole world each time. I've left this for when/if we do a larger rewrite to fix the other issues we're having...